core/src/main/java/org/javarosa/core/model/actions/Action.java

```diff
@@ -48,4 +44,22 @@ public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOExcep
     public void writeExternal(DataOutputStream out) throws IOException {
         ExtUtil.writeString(out, name);
     }
+
+    // Events that can trigger an action
+    public static final String EVENT_XFORMS_READY = "xforms-ready";
+    public static final String EVENT_XFORMS_REVALIDATE = "xforms-revalidate";
+    public static final String EVENT_JR_INSERT = "jr-insert";
+    public static final String EVENT_QUESTION_VALUE_CHANGED = "xforms-value-changed";
+
+    public static boolean isValidEvent(String s)  {
+        String[] allEvents = new String[] {EVENT_JR_INSERT,
+                EVENT_QUESTION_VALUE_CHANGED, EVENT_XFORMS_READY, EVENT_XFORMS_REVALIDATE};
+        for (String event : allEvents) {
+            if (event.equals(s)) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
I would make `allEvents` a `private static final Vector<String>` so that it is only constructed one time, not every time `isValidEvent` is called.
Then this can be `isValidEvent(String actionEventAttr) { return allEvents.contains(actionEventAttr); }`